src/server.py

```diff
@@ -118,26 +123,40 @@ def _ensure_release_countdown(self):
         self._countdown_id = GLib.timeout_add_seconds(
             self._TIMEOUT_S, self.release, priority=GLib.PRIORITY_LOW)
 
-    def play_sound(self, sound_event_id, connection, sender, path, iface,
-                   invocation):
+    def _play_sound(self, sound_event_id, connection, sender, path, iface,
+                   invocation, options=None):
         if sound_event_id not in self.metadata:
             invocation.return_dbus_error(
                 self._DBUS_UNKNOWN_SOUND_EVENT_ID,
                 "sound event with id %s does not exist" % sound_event_id)
             return
 
+        if options is None:
+            options = {}
+
         self._cancel_countdown()
         self.hold()
 
         uuid_ = str(uuid.uuid4())
-        self.players[uuid_] = HackSoundPlayer(self.metadata[sound_event_id],
-                                              sender)
+        metadata = self.metadata[sound_event_id].copy()
+        metadata.update(options)
```

Comment:
We can't do it this way; if, for example, a pitch of 1.5 is specified in the metadata file, and we specify pitch 1.06 in the options, then we want the pitch to become higher, not lower.

Volumes and pitches should be multiplied, delays should be added, and fade-in/fade-out times should probably be overridden.

Reply:
Should that multiplications be accumulated? If you call 3 times PlayFull with a pitch 1.06, should the pitch be multiplied by 1.06 * 1.06 * 1.06 in the third call?

Reply:
No, the parameters in PlayFull should only take effect for that one instance of playing the sound. So if you call it three times with pitches of 1.5, 1.4, 1.3, then the pitches should get subsequently lower, not higher.